src/main/java/com/openlattice/data/DataApi.java

```diff
@@ -38,8 +38,9 @@
     String BASE                  = SERVICE + CONTROLLER;
     // @formatter:on
 
-    String ASSOCIATION = "association";
+    String ASSOCIATION           = "association";
 
+    int    MAX_BATCH_SIZE        = 500;
```

Comment:
Why just 500?

Reply:
Since we don't have any request data (that I can find at least), I just picked something totally arbitrary. Happy to make it higher if you think 500 is too low

Reply:
I think the limit can be way higher -- the default upload batch size in shuttle is 100000 (although that doesn't end up turning into a single query, but I think that's still under the postgres query size threshold)

Reply:
Okay cool, I'll bump it up

Reply:
I think postgres can handle more, but I also can't say how much or how we use this request usually as you pointed out.
Maybe generate some fake queries and run EXPLAIN ANALYZE on prod with different number of UUIDs?

Reply:
I would very much like to do this but I don't think I have access yet :(
I do think some of those delete path postgres queries might be able to be optimized though

Reply:
Yeah, me neither, you have to go through some heavy background check to be able to access it.
@katherinebernstein : do you think it is worth to figure out a more precise threshold? Since you're the one who has access

Reply:
I'm pretty sure postgres can handle many millions of ids in a query

Reply:
My impression is that the `IN <array>` portion is suboptimal for a certain data scale. I imagine thousands is fine but the better thing would be to update that SQL if possible

Reply:
yeah true, avoiding bogging down postgres with huge queries is probably preferable

Reply:
The optimal / recommended way after a certain scale is to create an indexed temporary table and the join syntax to perform the delete. The next best thing is to choose a reasonable chunk size, use kotlin sequence with chunking to window over a list of ids. 

Our app server will OOM before Postgres runs into trouble at current scale.